azerothcore/azerothcore-wotlk
# fix(Core/Conditions): Implement CONDITION_DAILY_QUEST_DONE

## CHANGES PROPOSED:
* Implement Player::IsDailyQuestDone

* Implement CONDITION_DAILY_QUEST_DONE

* Imported from TC


## ISSUES ADDRESSED:
- Ref https://github.com/azerothcore/azerothcore-wotlk/issues/1372



## TESTS PERFORMED:
- Builds



## HOW TO TEST THE CHANGES:

```sql
DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=19 AND `SourceGroup`=0 AND `SourceEntry`=40;
INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES 
(19, 0, 40, 0, 0, 43, 0, 11153, 0, 0, 0, 0, 0, '', '');
```` 
THIS IS A DUMMY SQL. MAKE SURE TO DELETE IT AFTER TESTING.

1. Apply SQL 
2. .go c id 241
and see that he will offer only one quest (out of 2)
3. .q add 11153 (This is a daily quest)
4. .q comp 11153
5. .q rew 11153
6. See that he now offers two quests

Tada PR Works :) 

## KNOWN ISSUES AND TODO LIST:



## Target branch(es):








 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
